pkg/scanfetcher/nexpose.go

```diff
@@ -66,6 +68,7 @@ func (n *NexposeClient) FetchScans(ctx context.Context, ts time.Time) ([]domain.
 		completedScan, err := scanResourceToCompletedScan(resource, ts)
 		switch err.(type) {
 		case nil:
+			stater.Count("scanComplete", 1)
```

Comment:
suggestion: Emitting the metrics here mixes the business and operational logic. We try to separate these wherever possible so the concerns are better separated. The metric you're emitting here has a clear case for being added as a decorator instead.

Notably, this stat is equivalent to a count of the elements returned by the function. The function is part of a struct that implements the `domain.ScanFetcher`. This gives us some good foundation to separate the metric from the code by creating a wrapper for the original function. It generally looks something like this:

```golang
type NexposeClient struct {
  // ...
}
func (*NexposeClient) FetchScans(ctx context.Context, ts time.Time) (domain.CompletedScan, error) {
 // THE ORIGINAL IMPLEMENTATION HERE, UNALTERED
}

type ReportingScanFetcher struct {
  Wrapped domain.ScanFetcher
  StatFn domain.StatFn
}
func (f *ReportingScanFetcher) FetchScans(ctx context.Context, ts time.Time) (domain.CompletedScan, error) {
  // Start by calling the wrapped ScanFetcher which, in our case, will almost always be the NexposeClient
  // implementation that does the original work.
 scans, err := f.Wrapped.FetchScans(ctx, ts)
  // Emit a metric of the count of scans.
  if err != nil {
    f.StatFn(ctx).Count("statname", len(scans))
  }
  // Proxy the original return values so everything keeps working as expected.
  return scans, err
}
```

The reason we do this is to prevent the need to modify any stable business logic if/when our operational needs change. For example, if we decide to remove this stat then we should be able to do so without changing a single line from the `FetchScans` method on the `NexposeClient`.

Reply:
Also, we can use our existing `http.client.timing.count` to get a count of events sent to benthos, which would mean we don't need to add this stat here. Here's what I used to check to see that using this will give the same results as this added stat: 
`A = data('http.client.timing.count', filter=filter('client_dependency', 'benthos') and filter('environment', 'adev-west2') and filter('micros_service_id', 'sec-nexpose-scan-notifier'), rollup='sum').sum(by=['status_code']).publish(label='nexpose')`